core/contextmenu_items.js

```diff
@@ -270,12 +270,230 @@ Blockly.ContextMenuItems.registerWorkspaceOptions_ = function() {
   Blockly.ContextMenuItems.registerDeleteAll();
 };
 
+/** Option to duplicate a block. */
+Blockly.ContextMenuItems.registerDuplicate = function() {
+  var duplicateOption = {};
```

Comment:
`/** @type {!Blockly.ContextMenuRegistry.RegistryItem} */`

For all of these.

Reply:
Added type for each item. For the above comment, the ContextMenu class only deals with ContextMenuOption type, not RegistryItem. ContextMenuOption still adds some fields that are required (like weight) but I don't think it will break except in strict mode... For now I'm still holding off on making that change because it affects other pieces such as workspace comment options, but I'm thinking about that in the background but for now wanted to get review on the rest of it :)

Reply:
Sounds good